panther_hardware_interfaces/src/motors_controller.cpp

```diff
@@ -0,0 +1,246 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <panther_hardware_interfaces/motors_controller.hpp>
+
+namespace panther_hardware_interfaces
+{
+
+MotorsController::MotorsController(
+  const CanOpenSettings & canopen_settings, const DrivetrainSettings & drivetrain_settings)
+: canopen_controller_(canopen_settings),
+  front_data_(drivetrain_settings),
+  rear_data_(drivetrain_settings),
+  roboteq_vel_cmd_converter_(drivetrain_settings),
+  pdo_feedback_timeout_(canopen_settings.pdo_feedback_timeout)
+{
+}
+
+void MotorsController::Initialize()
+{
+  try {
+    canopen_controller_.Initialize();
+  } catch (const std::runtime_error & e) {
+    throw e;
+  }
+}
+
+void MotorsController::Deinitialize() { canopen_controller_.Deinitialize(); }
+
+void MotorsController::Activate()
+{
+  // Activation procedure - it is necessary to first reset scripts, wait for a bit (1 second)
+  // and then send 0 commands for some time (also 1 second)
+
+  try {
+    canopen_controller_.GetFrontDriver()->ResetRoboteqScript();
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error(
+      "Front driver reset Roboteq script exception: " + std::string(e.what()));
+  }
+
+  try {
+    canopen_controller_.GetRearDriver()->ResetRoboteqScript();
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error(
+      "Rear driver reset Roboteq script exception: " + std::string(e.what()));
+  }
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+
+  try {
+    canopen_controller_.GetFrontDriver()->SendRoboteqCmdChannel1(0);
+    canopen_controller_.GetFrontDriver()->SendRoboteqCmdChannel2(0);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error("Front driver send 0 command exception: " + std::string(e.what()));
+  }
+  try {
+    canopen_controller_.GetRearDriver()->SendRoboteqCmdChannel1(0);
+    canopen_controller_.GetRearDriver()->SendRoboteqCmdChannel2(0);
+  } catch (const std::runtime_error & e) {
+    throw std::runtime_error("Rear driver send 0 command exception: " + std::string(e.what()));
+  }
+  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
```suggestion
  try {
    canopen_controller_.GetFrontDriver()->SendRoboteqCmdChannel1(0);
    canopen_controller_.GetFrontDriver()->SendRoboteqCmdChannel2(0);
  } catch (const std::runtime_error & e) {
    throw std::runtime_error("Front driver send 0 command exception: " + std::string(e.what()));
  }

  try {
    canopen_controller_.GetRearDriver()->SendRoboteqCmdChannel1(0);
    canopen_controller_.GetRearDriver()->SendRoboteqCmdChannel2(0);
  } catch (const std::runtime_error & e) {
    throw std::runtime_error("Rear driver send 0 command exception: " + std::string(e.what()));
  }

  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```
Add lines after try block